filters/ignore_new_executors.hpp

```diff
@@ -0,0 +1,56 @@
+#include <ctime>
+#include <memory>
+
+#include "messages/serenity.hpp"
+
+#include "serenity/executor_map.hpp"
+#include "serenity/serenity.hpp"
+
+#include "stout/option.hpp"
+
+#ifndef SERENITY_IGNORE_NEW_TASKS_HPP
+#define SERENITY_IGNORE_NEW_TASKS_HPP
+
+namespace mesos {
+namespace serenity {
+
+/**
+ * IgnoreNewTasksFilter ignores removes executors that run for less time than
+ * threshold.
+ * It's purpose is to cut away tasks that are warming up.
+ */
+class IgnoreNewExecutorsFilter : public Consumer<ResourceUsage>,
+                             public Producer<ResourceUsage> {
+ public:
+  explicit IgnoreNewExecutorsFilter(uint32_t _threshold = DEFAULT_THRESHOLD) :
+      threshold(_threshold),
+      executorTimestamps(new ExecutorMap<time_t>) {}
+
+  ~IgnoreNewExecutorsFilter() {}
+
+  IgnoreNewExecutorsFilter(const IgnoreNewExecutorsFilter& other) :
+       threshold(other.threshold) {}
+
+  Try<Nothing> consume(const ResourceUsage& usage) override;
+
+  /// Set #seconds when executor is considered too fresh.
+  void setThreshold(uint32_t _threshold) {
+    this->threshold = _threshold;
+  }
+
+ protected:
+  /// ctime function wrapped for mocking purposes.
+  inline virtual time_t GetTime(time_t* arg) {
```

Comment:
It's function wrapper. I don't think it's reasonable to change it's signature.
Also I wonder if maybe we should inject GetTime functor in constructor (with default that uses ctime), instead of implementing this function in every class.